src/main/java/echo/parser/Parser.java

```diff
@@ -0,0 +1,256 @@
+                    }
+
+                    try {
+                        dateTimeString = splitSlash[1].substring(splitSlash[1].indexOf(" ") + 1);
+                        localDateTime = LocalDateTime.parse(dateTimeString,
+                                DateTimeFormatter.ofPattern("yyyy-M-d HHmm"));
+
+                        if (command.equals(DeadlineCommand.COMMAND)) {
+                            return prepareDeadlineCommand(input, desc, localDateTime);
+                        } else {
+                            return prepareEventCommand(input, desc, localDateTime);
+                        }
+                    } catch (DateTimeParseException e) {
+                        throw new EchoException("Invalid format. Follow the <yyyy-mm-dd> <24hr time> format."
+                                + "\n" + "E.g. 2019-10-15 1800");
+                    }
+                }
+            case MarkCommand.COMMAND:
+            case UnmarkCommand.COMMAND:
+            case DeleteCommand.COMMAND:
+                if (splitSpace.length == 1) {
+                    // If second input (task number) is not specified
+                    throw new EchoException("Please specify the task number. Eg. mark 1");
+                }
+
+                // Parse second input into an int, throws NumberFormatException if it's not an int.
+                try {
+                    int taskIndex = Integer.parseInt(splitSpace[1]) - 1;
+                    if (command.equals(MarkCommand.COMMAND)) {
+                        return prepareMarkCommand(taskIndex);
+                    } else if (command.equals(UnmarkCommand.COMMAND)) {
+                        return prepareUnmarkCommand(taskIndex);
+                    } else {
+                        return prepareDeleteCommand(taskIndex);
+                    }
+                } catch (NumberFormatException nfe) {
+                    // Second input provided by user is not an int.
+                    throw new EchoException("Second input must be an integer. Eg. mark 1");
+                }
+            case FindCommand.COMMAND:
+                if (splitSpace.length == 1) {
+                    // If second input (description) is not specified
+                    throw new EchoException("Please specify the description to find!");
+                }
+                return prepareFindCommand(input.substring(command.length() + 1).trim());
+            case ByeCommand.COMMAND:
+                return prepareByeCommand();
+            default:
+                return prepareHelpCommand();
+            }
+        } catch (ArrayIndexOutOfBoundsException e) {
+            // Deals with input that is just white space.
+            throw new EchoException("Invalid input!!!");
+        }
+    }
+
+    /**
+     * Is user exited.
+     *
+     * @return True if user is exited; False otherwise.
+     */
+    public static boolean isExit() {
+        return isExit;
+    }
+
+    /**
+     * Prepare list command.
+     *
```

Comment:
According to Java coding standard, the first sentence in method header comments should start in the form `Returns ...`, `Adds ...` (not `Return` or `Returnning` etc.).
Maybe you can consider changing it to better adhere to the standard? I noticed this same issue in several other places too.
```suggestion
    /**
     * Prepares list command.
     *
```

Reply:
Good eye! I will change it as you suggested 👍.